pccs/Dockerfile

```diff
@@ -0,0 +1,71 @@
+FROM ubuntu:20.04 AS builder
+
+# DCAP version (github repo branch, tag or commit hash)
+ARG DCAP_VERSION=DCAP_1.10.3
+
+# update and install packages
+RUN DEBIAN_FRONTEND=noninteractive \
+    apt-get update -yq && \
+    apt-get upgrade -yq && \
+    apt-get install -yq --no-install-recommends \
+    build-essential \
+    ca-certificates \
+    curl \
+    git \
+    zip
+
+# install node.js
+RUN curl -sL https://deb.nodesource.com/setup_14.x | bash -
+RUN DEBIAN_FRONTEND=noninteractive apt-get install -yq --no-install-recommends nodejs
+
+RUN apt-get clean && rm -rf /var/lib/apt/lists/*
+
+# clone DCAP repo
+RUN git clone https://github.com/intel/SGXDataCenterAttestationPrimitives.git -b ${DCAP_VERSION} --depth 1
+
+# remove conversion to hex
+RUN sed -i "s/result[['pckcrl']*] = Buffer.from(result[['pckcrl']*], 'utf8').toString('hex');/\/\/ EDIT: Removed line/" SGXDataCenterAttestationPrimitives/QuoteGeneration/pccs/services/pckcrlService.js
+RUN sed -i "s/crl = Buffer.from(crl, 'utf8').toString('hex');/\/\/ EDIT: Removed line/" SGXDataCenterAttestationPrimitives/QuoteGeneration/pccs/services/rootcacrlService.js
+
+# set PWD to PCKCertSelection dir
+WORKDIR /SGXDataCenterAttestationPrimitives/tools/PCKCertSelection/
+
+# build libPCKCertSelection library and copy to lib folder
+RUN make && \
+    mkdir -p ../../QuoteGeneration/pccs/lib && \
+    cp ./out/libPCKCertSelection.so ../../QuoteGeneration/pccs/lib/ && \
+    make clean
+
+# set PWD to PCCS dir
+WORKDIR /SGXDataCenterAttestationPrimitives/QuoteGeneration/pccs/
+
+# create cerificate
+RUN mkdir -p ./ssl_key && cd ssl_key && openssl genrsa -out private.pem 2048 && openssl req -new -key private.pem -out csr.pem -subj '/C=ES/ST=NRW/L=Bochum/O=ES/CN=es.com' && openssl x509 -req -days 365 -in csr.pem -signkey private.pem -out file.crt && rm -rf csr.pem && chmod 644 ./*
```

Comment:
I thought it would be cleaner to do it in the Dockerfile. Otherwise I have to install the openssl libraries in the resulting docker image, which are not used later on, right?

Reply:
We should not distribute things that contain private keys. Every user of the pccs image will then have the same key.
The openssl package isn't that large, so I think it's okay to install it.